hooks/useAvalancheForecast.ts

```diff
@@ -44,8 +32,9 @@ export const useAvalancheForecast = (center_id: string, forecast_zone_id: number
             url,
           },
         });
+      } else {
+        return parseResult.data;
       }
-      return productSchema.parse(data);
```

Comment:
I think we should be returning `parseResult.error` from the first branch of the `if` statement

Reply:
I don't know if it's just my editor, but boy are these types fun:

```
TS2345: Argument of type '() => Promise<{ status?: ProductStatus.Published; id?: number; media?: { type?: MediaType; id?: number; url?: any; caption?: string; }[]; expires_time?: string; published_time?: string; product_type?: ProductType; ... 11 more ...; forecast_zone?: { ...; }[]; } | ZodError<...>>' is not assignable to parameter of type 'QueryFunction<{ status?: ProductStatus.Published; id?: number; media?: { type?: MediaType; id?: number; url?: any; caption?: string; }[]; expires_time?: string; published_time?: string; product_type?: ProductType; ... 11 more ...; forecast_zone?: { ...; }[]; }, QueryKey>'.   Type 'Promise<{ status?: ProductStatus.Published; id?: number; media?: { type?: MediaType; id?: number; url?: any; caption?: string; }[]; expires_time?: string; published_time?: string; product_type?: ProductType; ... 11 more ...; forecast_zone?: { ...; }[]; } | ZodError<...>>' is not assignable to type '{ status?: ProductStatus.Published; id?: number; media?: { type?: MediaType; id?: number; url?: any; caption?: string; }[]; expires_time?: string; published_time?: string; product_type?: ProductType; ... 11 more ...; forecast_zone?: { ...; }[]; } | Promise<...>'.     Type 'Promise<{ status?: ProductStatus.Published; id?: number; media?: { type?: MediaType; id?: number; url?: any; caption?: string; }[]; expires_time?: string; published_time?: string; product_type?: ProductType; ... 11 more ...; forecast_zone?: { ...; }[]; } | ZodError<...>>' is not assignable to type 'Promise<{ status?: ProductStatus.Published; id?: number; media?: { type?: MediaType; id?: number; url?: any; caption?: string; }[]; expires_time?: string; published_time?: string; product_type?: ProductType; ... 11 more ...; forecast_zone?: { ...; }[]; }>'.       Type '{ status?: ProductStatus.Published; id?: number; media?: { type?: MediaType; id?: number; url?: any; caption?: string; }[]; expires_time?: string; published_time?: string; product_type?: ProductType; ... 11 more ...; forecast_zone?: { ...; }[]; } | ZodError<...>' is not assignable to type '{ status?: ProductStatus.Published; id?: number; media?: { type?: MediaType; id?: number; url?: any; caption?: string; }[]; expires_time?: string; published_time?: string; product_type?: ProductType; ... 11 more ...; forecast_zone?: { ...; }[]; }'.         Type 'ZodError<{ status?: ProductStatus.Published; id?: number; media?: { type?: MediaType; id?: number; url?: string | { large?: string; medium?: string; original?: string; thumbnail?: string; }; caption?: string; }[]; ... 14 more ...; forecast_zone?: { ...; }[]; }>' has no properties in common with type '{ status?: ProductStatus.Published; id?: number; media?: { type?: MediaType; id?: number; url?: any; caption?: string; }[]; expires_time?: string; published_time?: string; product_type?: ProductType; ... 11 more ...; forecast_zone?: { ...; }[]; }'.
```

Need to find the right type to parameterize the `useQuery`?

Reply:
Ah, I see - it's the difference between returning the error and throwing it. The other call (`const {data} = await axios.get(url);`) throws an `AxiosError` - and we have error-handling code in the calling components as well, so WDYT about not using `safeParse` here at all?

I think that would help us with #23 - perhaps we can add helpers for the Sentry bits since we want to associate extra data with these exceptions?

Reply:
NVM - we can just `throw` here and keep the flows the same.